In [1]:
import numpy as np
import pandas as pd
from sklearn.metrics import cohen_kappa_score

In [2]:
# Data rata-rata
ugm_avg = [673.08, 894.40, 695.21, 513.83, 704.32, 604.06, 415.76, 382.25, 484.65, 506.56, 286.87, 391.18, 498.08, 412.22]
undip_avg = [159.83, 273.33, 179.97, 161.96, 190.19, 140.42, 149.02, 162.29, 202.61, 191.17, 189.75, 149.70, 187.24, 159.66]


In [3]:
# Kategorisasi data (misalnya: 0 = rendah, 1 = sedang, 2 = tinggi)
def categorize(value):
    if value < 300:
        return 0  # Rendah
    elif value < 600:
        return 1  # Sedang
    else:
        return 2  # Tinggi

In [4]:
ugm_cat = [categorize(val) for val in ugm_avg]
undip_cat = [categorize(val) for val in undip_avg]

In [5]:
# Hitung Cohen's Kappa
kappa_score = cohen_kappa_score(ugm_cat, undip_cat)
kappa_score

0.0

# Two Sample T Test Independent

In [7]:
from scipy import stats

In [8]:
t_stat, p_value = stats.ttest_ind(ugm_avg,undip_avg)

# Interpretasi hasil
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

T-statistic: 7.954831512235475
P-value: 1.9667240825930664e-08


Penarikan kesimpulan

In [9]:
if p_value < 0.05:
    print("Terdapat perbedaan yang signifikan antara dua sampel.")
else:
    print("Tidak terdapat perbedaan yang signifikan antara dua sampel.")

Terdapat perbedaan yang signifikan antara dua sampel.


# Uji Kontrol

In [10]:
import pandas as pd
from scipy import stats

# Data 
data = {
    'Kode': ['K.1.1', 'K.1.2', 'K.1.3', 'K.1.4', 'K.1.5', 'K.1.6', 'K.1.7', 
             'K.2.1', 'K.2.2', 'K.2.3', 'K.2.4', 'K.2.5', 'K.2.6', 'K.2.7'],
    '1': [600.53, 862.69, 583.29, 574.23, 746.28, 799.3, 357.23, 454.16, 436.1, 593.63, 304.2, 317.94, 352.23, 252.35],
    '2': [403.5, 876.15, 773.07, 502.36, 817.3, 420.22, 352.03, 377.88, 423.42, 541.82, 332.06, 338.17, 509.51, 433.09],
    '3': [1008.08, 1018.05, 633.93, 484.48, 820.35, 587.18, 477.03, 372.88, 540.07, 500.21, 339.53, 306.76, 549.09, 508.7],
    '4': [991.27, 1017.38, 741.45, 456.98, 643.23, 759.19, 440.37, 313.6, 585.94, 527.12, 244.28, 391.18, 614.36, 454.06],
    '5': [616.06, 765.62, 794.45, 528.8, 695.39, 443.96, 445.48, 330.65, 416.28, 415.34, 212.57, 445, 495.81, 515.94],
    '6': [419.05, 826.49, 645.08, 536.12, 503.34, 614.48, 422.4, 444.3, 506.06, 461.23, 288.6, 548.05, 467.49, 309.17]
}

# Convert to DataFrame
df = pd.DataFrame(data)

# Calculate the average for each row
df['Average'] = df[['1', '2', '3', '4', '5', '6']].mean(axis=1)

# Separate control and treatment groups
control_group = df.loc[0:6, 'Average']
treatment_group = df.loc[7:13, 'Average']

# Perform independent t-test
t_stat, p_val = stats.ttest_ind(control_group, treatment_group)

t_stat, p_val

(3.375210498028608, 0.005516796396392764)

t-statistik: Nilai t-statistik sebesar 3.375 menunjukkan besarnya perbedaan relatif terhadap variasi dalam data sampel. Nilai absolut yang lebih besar menunjukkan perbedaan yang lebih signifikan antara dua kelompok. <br>

nilai-p: Nilai p sebesar 0.0055 lebih kecil dari tingkat signifikansi yang umum digunakan yaitu 0.05. Ini berarti terdapat perbedaan yang signifikan secara statistik antara kelompok kontrol (K1) dan kelompok perlakuan (K2).

In [11]:
# New dataset
data_new = {
    'Kode': ['K.1.1', 'K.1.2', 'K.1.3', 'K.1.4', 'K.1.5', 'K.1.6', 'K.1.7', 
             'K.2.1', 'K.2.2', 'K.2.3', 'K.2.4', 'K.2.5', 'K.2.6', 'K.2.7'],
    '1': [204.66, 296.2, 190.22, 174.7, 216.33, 141.25, 147.08, 176.35, 200.33, 195.09, 174.18, 119.19, 183.14, 145.19],
    '2': [118.16, 300.07, 188.7, 170.77, 201.94, 143.79, 151.18, 182.72, 205.54, 192, 182.9, 122.6, 178.36, 150.28],
    '3': [112.65, 315.46, 183.31, 140.42, 161.12, 136.22, 148.8, 152.67, 201.96, 186.43, 182.06, 127.24, 175.14, 141.8],
    '4': [122.43, 228.67, 201.33, 154.6, 154.6, 136.22, 148.8, 155.72, 201.96, 186.43, 182.06, 134.68, 173.22, 145.91],
    '5': [203.26, 239.82, 177.31, 222.33, 222.33, 154.6, 147.31, 147.31, 201.96, 186.43, 182.06, 134.68, 173.22, 145.91],
    '6': [197.83, 259.73, 138.95, 138.95, 138.95, 136.22, 148.8, 155.72, 201.96, 186.43, 182.06, 134.68, 173.22, 145.91]
}

# Convert to DataFrame
df_new = pd.DataFrame(data_new)

# Calculate the average for each row
df_new['Average'] = df_new[['1', '2', '3', '4', '5', '6']].mean(axis=1)

# Separate control and treatment groups
control_group_new = df_new.loc[0:6, 'Average']
treatment_group_new = df_new.loc[7:13, 'Average']

# Perform independent t-test
t_stat_new, p_val_new = stats.ttest_ind(control_group_new, treatment_group_new)

t_stat_new, p_val_new

(0.5049194684278471, 0.6227597455164491)

t-statistik: Nilai t-statistik sebesar 0.505 menunjukkan besarnya perbedaan relatif terhadap variasi dalam data sampel. Nilai absolut yang kecil menunjukkan perbedaan yang tidak signifikan antara dua kelompok. <br>

nilai-p: Nilai p sebesar 0.623 lebih besar dari tingkat signifikansi yang umum digunakan yaitu 0.05. Ini berarti tidak terdapat perbedaan yang signifikan secara statistik antara kelompok kontrol (K1) dan kelompok perlakuan (K2).

In [12]:
df_new

Kode       1       2       3       4       5       6     Average
0   K.1.1  204.66  118.16  112.65  122.43  203.26  197.83  159.831667
1   K.1.2  296.20  300.07  315.46  228.67  239.82  259.73  273.325000
2   K.1.3  190.22  188.70  183.31  201.33  177.31  138.95  179.970000
3   K.1.4  174.70  170.77  140.42  154.60  222.33  138.95  166.961667
4   K.1.5  216.33  201.94  161.12  154.60  222.33  138.95  182.545000
5   K.1.6  141.25  143.79  136.22  136.22  154.60  136.22  141.383333
6   K.1.7  147.08  151.18  148.80  148.80  147.31  148.80  148.661667
7   K.2.1  176.35  182.72  152.67  155.72  147.31  155.72  161.748333
8   K.2.2  200.33  205.54  201.96  201.96  201.96  201.96  202.285000
9   K.2.3  195.09  192.00  186.43  186.43  186.43  186.43  188.801667
10  K.2.4  174.18  182.90  182.06  182.06  182.06  182.06  180.886667
11  K.2.5  119.19  122.60  127.24  134.68  134.68  134.68  128.845000
12  K.2.6  183.14  178.36  175.14  173.22  173.22  173.22  176.050000
13  K.2.7  145.19  150.28  141.80  145.91  145.91  145.91  145.833333